File: CS50/sentimental-readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    letters = letter(text)
    words = word(text)
    sentences = sentence(text)

    L = letters / words * 100
    S = sentences / words * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def letter(text):
    count = 0

    for i in text:
        if i.isalpha():
            count += 1

    return count


def word(text):
    count = 1

    for i in text:
        if i == " ":
            count += 1

    return count


def sentence(text):
    count = 0

    for i in text:
        if i == "." or i == "!" or i == "?":
            count += 1

    return count


main()

File: CS50/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

double calculate_index(int letters, int words, int sentences);



int main(void)
{
    int letters;
    int words;
    int sentences;
    int index;

    string text = get_string("Text: ");

    letters = count_letters(text);                                  // Counting letters
    words = count_words(text);                                      // Counting words
    sentences = count_sentences(text);                              // Counting sentences

    index = calculate_index(letters, words, sentences);             // Calculating index

    if(index < 1)
        printf("Before Grade 1\n");
    else if (index >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %d\n",index);
}



// Counting letters
int count_letters(string text)
{
    int i = 0;
    int letters =0;

    while(text[i] != '\0')
    {
        if((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
            letters++;
        i++;
    }

    return letters;
}

// Counting words
int count_words(string text)
{
    int i = 0;
    int words =0;

    while(text[i] != '\0')
    {
        if(text[i] == ' ')
            words++;
        i++;
    }

    return words+1;
}

// Counting sentences
int count_sentences(string text)
{
    int i = 0;
    int sentences =0;

    while(text[i] != '\0')
    {
        if(text[i] == '.' || text[i] == '!' || text[i] == '?')
            sentences++;
        i++;

    }

    return sentences;
}

// Coleman-Liau index
double calculate_index(int letters, int words, int sentences)
{
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    return round(((0.0588 * L) - (0.296 * S) - 15.8));
}
